controller/appcontroller.go

```diff
@@ -2315,6 +2320,33 @@ func (ctrl *ApplicationController) newApplicationInformerAndLister() (cache.Shar
 	if err != nil {
 		return nil, nil
 	}
+
+	// Periodically updates ObservedAt time in Application health status to indicate freshness
+	_, err = informer.AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			UpdateFunc: func(old, new interface{}) {
+				if !ctrl.canProcessApp(new) {
+					return
+				}
+				oldApp, oldOK := old.(*appv1.Application)
+				newApp, newOK := new.(*appv1.Application)
+				if oldOK && newOK {
+					// To prevent a status update hot loop, enqueue only events generated by resyncs
+					// and ignore those triggered by application updates, which are handled by another event handler.
+					if oldApp.ResourceVersion != newApp.ResourceVersion {
+						return
+					}
+
+					// We are interested only in refreshing the status, not performing a resource-level comparison.
+					ctrl.requestAppRefresh(newApp.QualifiedName(), ComparisonWithNothing.Pointer(), nil)
+				}
+			},
+		},
+		appHealthStatusResyncDuration, // 30s
+	)
+	if err != nil {
+		return nil, nil
```

Comment:
Is the error actually safe to ignore? Maybe at least log it?

Reply:
I'm not entirely sure, but I followed the existing code to set up the handler from the controller where the error is currently being ignored. I looked into the [informer.AddEventHandlerWithResyncPeriod](https://github.com/kubernetes/client-go/blob/c5196ebcc18e1bf29561b7a689fdb121913d221b/tools/cache/shared_informer.go#L581) func, it seems the only error it returns is when the shared informer is stopped. Based on my understanding, it seems safe to ignore the error. I have also added a log statement as suggested.